src/components/review/data-list/page-client.tsx

```diff
@@ -0,0 +1,127 @@
+'use client';
+
+import React, { useState, useCallback } from 'react';
+import { Card, CardContent, CardHeader, CardTitle } from '@/components/ui/card';
+import { Filters } from './filters';
+import { ReviewTable } from './table';
+import { LoadingSkeleton } from './loading-skeleton';
+import { useDateRangeValues } from '@/lib/shared/utils/date-range';
+import { useMonthlySummaryQuery } from '@/lib/review';
+import type { DateRangeOption } from '@/lib/shared/utils/date-range';
+import { AlertCircle } from 'lucide-react';
+import { Alert, AlertDescription } from '@/components/ui/alert';
+
+export function ReviewDataListPageClient() {
+  // Local state for filters
+  const [dateRange, setDateRange] = useState<DateRangeOption>('all-time');
+  const [selectedDistricts, setSelectedDistricts] = useState<string[]>([]);
+  const [appliedFilters, setAppliedFilters] = useState<{
+    dateFrom?: string;
+    dateTo?: string;
+    districts?: string[];
+  }>({});
+
+  const { createdAfter, createdBefore } = useDateRangeValues(dateRange);
+
+  // Convert ISO dates to YYYY-MM-DD format for API
+  const dateFrom = createdAfter
+    ? new Date(createdAfter).toISOString().split('T')[0]
+    : undefined;
+  const dateTo = createdBefore
+    ? new Date(createdBefore).toISOString().split('T')[0]
+    : undefined;
+
+  // Fetch monthly summary data with applied filters
+  const { data, isLoading, isFetching, error } = useMonthlySummaryQuery({
+    dateFrom: appliedFilters.dateFrom,
+    dateTo: appliedFilters.dateTo,
+    districts: appliedFilters.districts,
+  });
+
+  const summaries = data?.monthlySummaries ?? [];
+  const availableDistricts = data?.availableDistricts ?? [];
+
+  const handleDateRangeChange = useCallback((newDateRange: DateRangeOption) => {
+    setDateRange(newDateRange);
+  }, []);
+
+  const handleDistrictsChange = useCallback((districts: string[]) => {
+    setSelectedDistricts(districts);
+  }, []);
+
+  const handleApplyFilters = useCallback(() => {
+    setAppliedFilters({
+      dateFrom,
+      dateTo,
+      districts: selectedDistricts.length > 0 ? selectedDistricts : undefined,
+    });
+  }, [dateFrom, dateTo, selectedDistricts]);
+
+  const handleReview = useCallback((district: string, month: string) => {
+    // This will be handled by the Table component
+    console.log('Review clicked for:', district, month);
+  }, []);
+
+  if (error) {
```

Comment:
The good news is that our setup of the Tanstack Query Client keeps track of the error state and displays it to the user as a toaster, so you dont need to separately keep track of errors